### ISSUE TITLE
Reapplications: write and run script

### ISSUE THREAD

**Issue by dommitchell** [DOAJ/doaj#1376]
Before we can publicise the end of the processing of the reapplications, we need to do some housekeeping. To help us, we would like you to create the following:

1) a list of journals still In DOAJ which have a rejected reapplication.
2) a list of journals In DOAJ with no reapplication record and no Tick
3) a list of journals In DOAJ that have the note: 'Continuation automatically extracted from journal {x} during migration'
[NOT NEEDED 4) a list of journals In DOAJ without The Tick]

Each report should be importable in Excel etc and include: Title, ISSN and internal DOAJ ID.

I expect that there are some stragglers in DOAJ that have the same title but different ISSNs (due to being continuations) so the comparisons for (1) and (2) should be run against ISSN and title.

I expect the 4th one of these will consist largely of journals already covered by other lists but we want to catch all eventualities.

We would like this to be given highest priority so that we can announce the end of the Reapps process.

**Comment by Steven-Eardley** [comment]
The approach here will be to scroll through all of the journals with ```in_doaj = true``` once, with the 4 report generators gathering information from each journal. They will create 1 CSV file each, with the columns as above except that ISSN will be split into two columns for PISSN and EISSN.

> 1. a list of journals still In DOAJ which have a rejected reapplication.

Given the journal, check ```admin.current_application``` to get the reapplication record ID, verify that it's a reapplication (```index.application_type```) then add to report if it's in ```rejected``` status.

> 2. a list of journals In DOAJ with no reapplication record
Given the journal, check ```admin.current_application``` to get the reapplication record ID. If there isn't one, add it to the report.

> 3. a list of journals In DOAJ that have the note: 'Continuation automatically extracted from journal {x} during migration'
Given the journal, perform a text (regex) search on the notes field. Add to report if present.

> 4. a list of journals In DOAJ without The Tick
Given the journal, add to report if ```journal.is_ticked()``` is False.

The rows in the CSV files will be initially ordered by DOAJ ID, but your spreadsheet program can sort by title or by ISSN to deal with continuations.

**Comment by dommitchell** [comment]
> Given the journal, check admin.current_application to get the reapplication record ID, verify that it's a reapplication (index.application_type) then add to report if it's in rejected status.
> 
> a list of journals In DOAJ with no reapplication record

Steve, sorry. I need to modify this one:

- a list of journals In DOAJ with no reapplication record and no Tick.

**Comment by Steven-Eardley** [comment]
That makes sense, I’ll include that. I hadn’t written that one yet, so no effort wasted!

**Comment by dommitchell** [comment]
Hi Steve, if I'm not too late, I don't need (4) as I have that information already in the journal csv.

**Comment by Steven-Eardley** [comment]
Already done, but it took only a few minutes once I'd written the others. I've tested this script on the test server, awaiting responses from m'colleagues.

**Comment by Steven-Eardley** [comment]
@dommitchell I went ahead and verified it again myself, then ran it on live for you. You'll have the email, but the resulting reports are also here:

[reapplication_report_2017-10-13.zip](https://github.com/DOAJ/doaj/files/1382429/reapplication_report_2017-10-13.zip)

**Comment by dommitchell** [comment]
Fab! Thanks. Have seen the files. Closing this for now.
